R/args.R

```diff
@@ -560,68 +587,70 @@ maybe_generate_seed <- function(seed, num_runs) {
   seed
 }
 
-
-# Check if metric speficied as file(s).
-# This particular function doesn't check if files exist
-metric_is_file <- function(metric) {
-  if (is.null(metric)) return(FALSE)
-  if (length(metric) > 1) return(TRUE)
-  !metric %in% available_metrics()
-}
-
 #' Validate metric
 #' @noRd
 #' @param metric User's `metric` argument.
 #' @param num_runs Number of CmdStan runs (number of MCMC chains).
 #' @return Either throws an error or returns `invisible(TRUE)`.
 #'
-validate_metric <- function(metric, num_runs) {
+validate_metric <- function(metric) {
   if (is.null(metric)) {
     return(invisible(TRUE))
   }
 
   checkmate::assert_character(metric, any.missing = FALSE, min.len = 1)
 
-  # TODO: need to check if in the right format (see CmdStanPy implementation)
-  if (length(metric) == 1) {
-    must_have_file <- !metric %in% available_metrics()
-    if (must_have_file) {
-      if (!checkmate::test_file_exists(metric, access = "r")) {
-        stop("'metric' is not one of {'diag_e', 'dense_e', 'unit_e'} but ",
-             "is also not a path to a readable file.", call. = FALSE)
-      }
+  return(invisible(metric %in% available_metrics()))
+}
+
+#' Validate metric file
+#' @noRd
+#' @param metric_file User's `metric_file` argument.
+#' @param num_runs Number of CmdStan runs (number of MCMC chains).
+#' @return Either throws an error or returns `invisible(TRUE)`.
+#'
+validate_metric_file <- function(metric_file, num_runs) {
+  if (is.null(metric_file)) {
+    return(invisible(TRUE))
+  }
+
+  checkmate::assert_character(metric_file, any.missing = FALSE, min.len = 1)
```

Comment:
I think `min.len=1` will cause this to fail every time `metric_file` isn't specified! (`NULL is length 0)

Reply:
I think this is one reason why tests are failing